src/core.jl

```diff
@@ -151,10 +151,16 @@ function TensorShape(dims::Vector)
     TensorShape([x<0 ? Nullable{Int64}() : Nullable{Int64}(x) for x in dims])
 end
 
-function TensorShape(dim::Void)
+function TensorShape(::typeof(collect(tuple())))
```

Comment:
What does this complicated type mean? Is there is a more compact type signature that would work?